Mage/src/main/java/mage/abilities/effects/common/continuous/SetBasePowerToughnessSourceEffect.java

```diff
@@ -72,15 +82,11 @@ public boolean apply(Game game, Ability source) {
             discard();
```

Comment:
There are custom duration, e.g. manual control of effect's lifecycle (direct `discard()` call). But I can't see any usage of ZCC related code like `getSourcePermanentIfItStillExists`. So it can be potentially affected by remove/blink bugs.

Reply:
Good find. I wasn't initially looking at the core logic, but I agree that zcc should be checked. I added it.